mathics/core/pattern.py

```diff
@@ -322,17 +322,22 @@ def per_expr(yield_expr, expr_groups, sum=0):
 
                         if expr_groups:
                             expr, count = expr_groups.popitem()
-                            max_per_pattern = count / len(patterns)
+                            max_per_pattern = count // len(patterns)
                             for per_pattern in range(max_per_pattern, -1, -1):
                                 for next in per_expr(   # nopep8
                                     expr_groups, sum + per_pattern):
                                     yield_expr([expr] * per_pattern + next)
                         else:
                             if sum >= match_count[0]:
                                 yield_expr([])
+                            # FIXME: Is this right?
+                            # For now we'll return basically no match.
+                            yield None
 
                     # for sequence in per_expr(expr_groups.items()):
                     def yield_expr(sequence):
+                        # FIXME: this call is wrong and needs a
```

Comment:
Is this difficult to fix?

Reply:
Short answer is yes. 

The needs a function is some sort of iterator over pattern groups. I don't understand what is desired here let alone how to write such a generator. It may be related to https://github.com/mathics/Mathics/issues/748